### ISSUE TITLE
Fedora: Opencv.so error

### ISSUE THREAD

**Issue by commet24** [WarmUpTill/SceneSwitcher#1129]
**Description**
This bug is on Fedora 39, when I try to open Advanced Scene Switcher in OBS. After unsuccessful launch OBS crashes.

**To Reproduce**
1. Use Fedora 39
2. Download Advanced Scene Switcher (tar.xz)
3. Extract all data to folders
4. Open tab in OBS

**Expected behavior**
Expected to run Advanced Scene Switcher

**Logs**
I don't understand Fedora's system of saving and sending logs. It was replaced by some crappy GUI where you can send error log somewhere.

**Version information**
 - OS: Fedora 39
 - OBS Version: 30.1.1
 - Plugin Version: 1.26.3

**Comment by WarmUpTill** [comment]
Sorry about that!

Just to clarify:
Did this work with a previous version of the plugin?
Is there _any_ log, since you hinted at the opencv.so lib being related to the issue in the title?

**Comment by WarmUpTill** [comment]
This might be a similar issue to https://github.com/WarmUpTill/SceneSwitcher/issues/1139.
The build artefacts were built in an Ubuntu 22.04 build environment, which might not be compatible with Fedora 39.
I can recommend trying the snap or flatpak version of OBS + the plugin instead:
`flatpak install com.obsproject.Studio.Plugin.SceneSwitcher`
`snap install obs-studio`

**Comment by mihawk90** [comment]
> I don't understand Fedora's system of saving and sending logs. It was replaced by some crappy GUI where you can send error log somewhere.

That's not true. Application logs are stored in the same place they always are. What you're referring to are coredumps (i.e. crashlogs) that *can* (but don't have to) be sent to the RedHat bugzilla. The logs from OBS are separate for one, and also doesn't keep you from running a debug on the coredump manually.

See here for GDB instructions:
https://gist.github.com/mihawk90/ef70c4f89e993f8139e8be911923078d

My guess is you are simply missing the required dependencies. Copying libraries doesn't copy the required dependencies after all and the OpenCV functions obviously have a dependency on OpenCV itself, `opencv-objdetect` and `opencv-stitching` at minimum.

---

Regardless of the above, I'm currently working on building an RPM for the plugin, which automatically pulls in said dependencies:
https://github.com/mihawk90/obs-studio-plugins.spec/blob/main/obs-studio-plugin-advanced-scene-switcher/obs-studio-plugin-advanced-scene-switcher.spec

No ready-made packages yet, still need to figure some things out.
